torch/nn/utils/weight_norm.py

```diff
@@ -34,8 +39,15 @@ def apply(module, name, dim):
         # remove w from parameter list
         del module._parameters[name]
 
+        # initial value for g
+        if init == 'default':
+            g_init = norm_except_dim(weight, 2, dim)
+        elif init == 'norm_preserving':
+            fan_in, fan_out = _calculate_fan_in_and_fan_out(weight)
+            g_init = ones_like(norm_except_dim(weight, 2, dim)) * math.sqrt(gamma * fan_in / fan_out)
```

Comment:
`full_like` will make this much faster.

Reply:
Thanks for the suggestion, I'll update the code to use `full_like`.